dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractAnalyticsService.java

```diff
@@ -539,12 +548,14 @@ private List<String> getDimensionItemUidList( EventQueryParams params, QueryItem
         {
             return item.getOptionSetFilterItemsOrAll();
         }
-        else
+        else if ( itemOptions != null )
```

Comment:
I think you could rewrite this Javadoc to something like:

    /**
     * Return a list of dimension item uids.
     * If itemOptions is null, it returns an empty List, which means no options set and no legend set.
     *
     * @param params EventQueryParams.
     * @param item QueryItem
     * @param itemOptions itemOtion list.
     * @return a list of uids.
     */